### ISSUE TITLE
Release 4.3.3 - Manual testing - `<client>` block re-definition

### ISSUE THREAD

**Issue by damarisg** [wazuh/wazuh-qa#2940]
# Description

The following issue aims to run the specified test for the current release candidate, report the results, and opens new issues.

| **Related Issue**| 
|---|
| https://github.com/wazuh/wazuh/issues/13583%7C 

It has been detected an error in agent enrollment under certains circunstances.
As @Vikman has specified in https://github.com/wazuh/wazuh/issues/13583#issuecomment-1141994629, agents fails to register to the manager using the following configuration

[`Client`](https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/client.html) is an agent's configuration related to the connection to the manager.  Regarding to this bug it seems
that the bug is produced when multiple `client` configuration block are used, making in this case that
the agent its ability to resolve FQDNs.

## Test information
|                                        |                                            |
|----------------------------|--------------------------------------------|
| **Test name**                 |   Client Configuration                                         |
| **Category**                   |     Enrollment 
 **Deployment option**   |    Installation                                        |
| **Main release issue**    |     https://github.com/wazuh/wazuh/issues/13583                                       |
| **Release candidate #** |    4.3.3                                      |
| **Agent Packages**      |   [Debian](https://packages-dev.wazuh.com/warehouse/pullrequests/4.3/deb/var/wazuh-agent_4.3.3-0.commitdd0c508_amd64.deb) - [RPM](https://packages-dev.wazuh.com/warehouse/pullrequests/4.3/rpm/var/wazuh-agent-4.3.3-0.commitdd0c508.x86_64.rpm) 
| **Manager Packages**  |   [Debian](https://packages-dev.wazuh.com/warehouse/pullrequests/4.3/deb/var/wazuh-manager_4.3.3-0.commitdd0c508_amd64.deb ) - [RPM](https://packages-dev.wazuh.com/warehouse/pullrequests/4.3/rpm/var/wazuh-manager-4.3.3-0.commitdd0c508.x86_64.rpm) 

# Automated Tests

- [x] Systems Test 
- [x] Integrations Test 
- [x] Registration Test 

# Uses Cases

### RPM

- [x] (TC_1.1): Resolve an FQDN and define enrollment in another block
- [x] (TC_2.1): Define the options in one block and continue with enrollment in another. The values must be maintained
- [x] (TC_3.1): Divide enrollment. All values must apply
- [x] (TC_4.1): Set up connection with two managers, one with IP and one with FQDN, with enrollment in between
- [x] (TC_5.1): Upgrade 4.3.2 to 4.3.3 - Resolve an FQDN and define enrollment in another block
- [x] (TC_6.1): Upgrade 4.3.2 to 4.3.3 - Define the options in one block and continue with enrollment in another. The values must be maintained.
- [x] (TC_7.1): Upgrade 4.3.2 to 4.3.3 - Divide enrollment. All values must apply
- [x] (TC_8.1): Upgrade 4.3.2 to 4.3.3 - Set up connection with two managers, one with IP and one with FQDN, with enrollment in between.

### DEB

- [x] (TC_1.2): Resolve an FQDN and define enrollment in another block
- [x] (TC_2.2): Define the options in one block and continue with enrollment in another. The values must be maintained
- [x] (TC_3.2): Divide enrollment. All values must apply
- [x] (TC_4.2): Set up connection with two managers, one with IP and one with FQDN, with enrollment in between
- [x] (TC_5.2): Upgrade 4.3.2 to 4.3.3 - Resolve an FQDN and define enrollment in another block
- [x] (TC_6.2): Upgrade 4.3.2 to 4.3.3 - Define the options in one block and continue with enrollment in another. The values must be maintained.
- [x] (TC_7.2): Upgrade 4.3.2 to 4.3.3 - Divide enrollment. All values must apply
- [x] (TC_8.2): Upgrade 4.3.2 to 4.3.3 - Set up connection with two managers, one with IP and one with FQDN, with enrollment in between.

### RPM

- [x] (TC1_1.1):  Upgrade 4.3.2 to 4.3.3 - Resolve an IP and define enrollment in another block
- [x] (TC1_2.1):  Set up connection with two managers, one with IP and one with manager_address enrollment 

### DEB

- [x] (TC1_1.2): Upgrade 4.3.2 to 4.3.3 - Resolve an IP and define enrollment in another block
- [x] (TC1_2.2):  Set up connection with two managers, one with IP and one with manager_address enrollment 

---

# Configuration - Tier 0

<details>
<summary><code>TC_1</code></summary>

```xml
<client>
    <server>
        <address>X</address>
    </server>
</client>

<client>
    <enrollment>
        <groups>default</groups>
    <enrollment>
</client>
```

</details>

<details>
<summary><code>TC_2</code></summary>

```xml
<client>
    <server><address>groovy</address></server>
    <notify_time>15</notify_time>
    <time-reconnect>45</time-reconnect>
    <force_reconnect_interval>120</force_reconnect_interval>
    <ip_update_interval>5</ip_update_interval>
    <auto_restart>no</auto_restart>
    <crypto_method>blowfish</crypto_method>
</client>

<client>
    <enrollment>
        <groups>default</groups>
    </enrollment>
</client>
```

</details>

<details>
<summary><code>TC_3</code></summary>

```xml
<client>
    <enrollment>
        <agent_name>Rock</agent_name>
    <enrollment>
</client>

<client>
    <server>
        <address>X</address>
    </server>
</client>

<client>
    <enrollment>
        <groups>default</groups>
    </enrollment>
</client>
```

</details>

<details>
<summary><code>TC_4</code></summary>

```xml
<client>
    <server>
        <address>X</address>
    </server>
</client>

<client>
    <enrollment>
        <groups>default</groups>
    </enrollment>
</client>

<client>
    <server>
        <address>192.168.33.12</address>
    </server>
</client>
```

</details>
<details>
<summary><code>TC_5</code></summary>

```xml
<client>
    <server>
        <address>X</address>
    </server>
</client>

<client>
    <enrollment>
        <groups>default</groups>
    <enrollment>
</client>
```
</details>

# Configuration Tier 1 

<details>
<summary><code>TC1_1</code></summary>

```xml
<client>
    <server>
        <address>IP</address>
    </server>
</client>

<client>
    <enrollment>
        <groups>default</groups>
    <enrollment>
</client>
```

</details>

<details>
<summary><code>TC1_2</code></summary>

```xml
  <client>
    <server>
      <address>manager1_IP</address>
      <port>1514</port>
      <protocol>tcp</protocol>
    </server>
    <config-profile>ubuntu, ubuntu20, ubuntu20.04</config-profile>
    <notify_time>10</notify_time>
    <time-reconnect>60</time-reconnect>
    <auto_restart>yes</auto_restart>
    <crypto_method>aes</crypto_method>
  </client>
  <client>
    <enrollment>
      <enabled>yes</enabled>
      <manager_address>manager2_IP</manager_address>
    </enrollment>
  </client>
```
</details>

**Comment by jmv74211** [comment]
## (TC_1): Resolve an FQDN and define enrollment in another block

<details>
<summary>Tester @Vikman - <code>Ubuntu / Sources / AddressSanitizer</code> :green_circle: </summary>

```
2022/05/31 14:59:14 wazuh-agentd: INFO: Server hostname resolved: groovy/192.168.33.22
```

</details>


<details>
<summary>Tester @Chema - <code>CentOS / Sources / Valgrind</code> :green_circle: </summary>

```
2022/05/31 17:50:41 wazuh-agentd[7548] start_agent.c:98 at connect_server(): INFO: Trying to connect to server (ubuntu/192.168.1.190:1514/tcp).
2022/05/31 17:50:41 wazuh-agentd[7548] start_agent.c:361 at agent_handshake_to_server(): INFO: (4102): Connected to the server (ubuntu/192.168.1.190:1514/tcp).
```

Valgrind

```
==7548== LEAK SUMMARY:
==7548==    definitely lost: 0 bytes in 0 blocks
==7548==    indirectly lost: 0 bytes in 0 blocks
==7548==      possibly lost: 1,088 bytes in 4 blocks
==7548==    still reachable: 50,939 bytes in 47 blocks
==7548==         suppressed: 0 bytes in 0 blocks
==7548== Reachable blocks (those to which a pointer was found) are not shown.
==7548== To see them, rerun with: --leak-check=full --show-leak-kinds=all
```

</details>


<details>
<summary>Tester @Marcel - <code>CentOS / Packages</code> :green_circle: </summary>

```
2022/05/31 16:10:57 wazuh-agentd: INFO: Server hostname resolved: centosmanager/10.2.0.10
2022/05/31 16:10:57 wazuh-agentd: INFO: Requesting a key from server: centosmanager/10.2.0.10
...
2022/05/31 16:11:17 wazuh-agentd: INFO: Trying to connect to server (centosmanager/10.2.0.10:1514/tcp).
2022/05/31 16:11:17 wazuh-agentd: INFO: (4102): Connected to the server (centosmanager/10.2.0.10:1514/tcp).
```

</details>

<details>
<summary>Tester @JuanCabrera - <code>DEB / Packages</code> :green_circle: </summary>

```
2022/05/31 16:15:38 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2022/05/31 16:15:38 wazuh-agentd: INFO: Using AES as encryption method.
2022/05/31 16:15:38 wazuh-agentd: INFO: Trying to connect to server (10.0.0.1:1514/tcp).
2022/05/31 16:15:38 wazuh-agentd: INFO: (4102): Connected to the server (10.0.0.1:1514/tcp).
```

</details>


## (TC_2): Define the options in one block and continue with enrollment in another. The values must be maintained


<details>
<summary>Tester @Vikman - <code>Ubuntu / Sources / AddressSanitizer</code> :green_circle: </summary>

```
{"data": {"client": {"notify_time": 15, "time-reconnect": 45, "force_reconnect_interval": 120, "ip_update_interval": 5, "auto_restart": "no", "remote_conf": "yes", "crypto_method": "blowfish", "server": [{"address": "impish/192.168.33.12", "port": 1514, "max_retries": 5, "retry_interval": 10, "protocol": "tcp"}], "enrollment": {"enabled": "yes", "delay_after_enrollment": 20, "port": 1515, "group": "default", "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH", "auto_method": "no"}}}, "error": 0}
```

</details>


<details>
<summary>Tester @Chema - <code>CentOS / Sources / Valgrind</code> :green_circle: </summary>

```
2022/05/31 18:02:09 wazuh-agentd: INFO: Trying to connect to server (ubuntu/192.168.1.190:1514/tcp).
2022/05/31 18:02:09 wazuh-agentd: INFO: (4102): Connected to the server (ubuntu/192.168.1.190:1514/tcp).
```

Valgrind

```
==7554== LEAK SUMMARY:
==7554==    definitely lost: 0 bytes in 0 blocks
==7554==    indirectly lost: 0 bytes in 0 blocks
==7554==      possibly lost: 1,088 bytes in 4 blocks
==7554==    still reachable: 51,977 bytes in 50 blocks
==7554==         suppressed: 0 bytes in 0 blocks
==7554== Reachable blocks (those to which a pointer was found) are not shown.
==7554== To see them, rerun with: --leak-check=full --show-leak-kinds=all
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client" -H "Authorization: Bearer $TOKEN"
{"data": {"client": {"notify_time": 15, "time-reconnect": 45, "force_reconnect_interval": 10, "ip_update_interval": 5, "auto_restart": "no", "remote_conf": "yes", "crypto_method": "blowfish", "server": [{"address": "ubuntu/192.168.1.190", "port": 1514, "max_retries": 5, "retry_interval": 10, "protocol": "tcp"}], "enrollment": {"enabled": "yes", "delay_after_enrollment": 20, "port": 1515, "group": "default", "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH", "auto_method": "no"}}}, "error": 0}
```

</details>


<details>
<summary>Tester @Marcel - <code>CentOS / Packages</code> :green_circle: </summary>

```
2022/05/31 16:23:03 wazuh-agentd: INFO: Server hostname resolved: centosmanager/10.2.0.10
2022/05/31 16:23:03 wazuh-agentd: INFO: Using Blowfish as encryption method.
2022/05/31 16:23:03 wazuh-agentd: INFO: Trying to connect to server (centosmanager/10.2.0.10:1514/tcp).
2022/05/31 16:23:03 wazuh-agentd: INFO: (4102): Connected to the server (centosmanager/10.2.0.10:1514/tcp).
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client?pretty=true" -H  "Authorization: Bearer $TOKEN"
{
   "data": {
      "client": {
         "notify_time": 15,
         "time-reconnect": 45,
         "force_reconnect_interval": 10,
         "ip_update_interval": 5,
         "auto_restart": "no",
         "remote_conf": "yes",
         "crypto_method": "blowfish",
         "server": [
            {
               "address": "centosmanager/10.2.0.10",
               "port": 1514,
               "max_retries": 5,
               "retry_interval": 10,
               "protocol": "tcp"
            }
         ],
         "enrollment": {
            "enabled": "yes",
            "delay_after_enrollment": 20,
            "port": 1515,
            "group": "default",
            "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH",
            "auto_method": "no"
         }
      }
   },
   "error": 0
}
```

</details>

<details>
<summary>Tester @JuanCabrera @Julia - <code>DEB / Packages</code> :green_circle: </summary>

```
2022/05/31 16:15:38 wazuh-agentd: INFO: (1410): Reading authentication keys file.
2022/05/31 16:15:38 wazuh-agentd: INFO: Using AES as encryption method.
2022/05/31 16:15:38 wazuh-agentd: INFO: Trying to connect to server (10.0.0.1:1514/tcp).
2022/05/31 16:15:38 wazuh-agentd: INFO: (4102): Connected to the server (10.0.0.1:1514/tcp).
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client?pretty=true" -H  "Authorization: Bearer $token"
{
   "data": {
      "client": {
         "config-profile": "ubuntu, ubuntu16, ubuntu16.04",
         "notify_time": 10,
         "time-reconnect": 60,
         "force_reconnect_interval": 0,
         "ip_update_interval": 0,
         "auto_restart": "yes",
         "remote_conf": "yes",
         "crypto_method": "aes",
         "server": [
            {
               "address": "10.0.0.1",
               "port": 1514,
               "max_retries": 5,
               "retry_interval": 10,
               "protocol": "tcp"
            }
         ],
         "enrollment": {
            "enabled": "yes",
            "delay_after_enrollment": 20,
            "port": 1515,
            "group": "default",
            "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH",
            "auto_method": "no"
         }
      }
   },
   "error": 0
}
```

</details>


## (TC_3): Define the options in one block and continue with enrollment in another. The values must be maintained


<details>
<summary>Tester @Vikman - <code>Ubuntu / Sources / AddressSanitizer</code> :green_circle: </summary>

```
001 Rocket any 500b25beaf3f00de13a82bf19a9c15bdd69708cea69f40b5ac18c038db897642
```

</details>


<details>
<summary>Tester @Chema - <code>CentOS / Sources / Valgrind</code> :green_circle: </summary>

```
2022/05/31 18:04:19 wazuh-agentd[9016] start_agent.c:98 at connect_server(): INFO: Trying to connect to server (ubuntu/192.168.1.190:1514/tcp).
2022/05/31 18:04:19 wazuh-agentd[9016] start_agent.c:361 at agent_handshake_to_server(): INFO: (4102): Connected to the server (ubuntu/192.168.1.190:1514/tcp).
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client" -H "Authorization: Bearer $TOKEN"
{"data": {"client": {"notify_time": 15, "time-reconnect": 45, "force_reconnect_interval": 10, "ip_update_interval": 5, "auto_restart": "no", "remote_conf": "yes", "crypto_method": "blowfish", "server": [{"address": "ubuntu/192.168.1.190", "port": 1514, "max_retries": 5, "retry_interval": 10, "protocol": "tcp"}], "enrollment": {"enabled": "yes", "delay_after_enrollment": 20, "port": 1515, "agent_name": "Rock", "group": "default", "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH", "auto_method": "no"}}}, "error": 0}
```

</details>


<details>
<summary>Tester @Marcel - <code>CentOS / Packages</code> :green_circle: </summary>

```
2022/05/31 16:44:41 wazuh-agentd: INFO: Trying to connect to server (10.2.0.10:1514/tcp).
2022/05/31 16:44:41 wazuh-agentd: INFO: (4102): Connected to the server (10.2.0.10:1514/tcp).
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client?pretty=true" -H  "Authorization: Bearer $TOKEN"
{
   "data": {
      "client": {
         "notify_time": 10,
         "time-reconnect": 60,
         "force_reconnect_interval": 0,
         "ip_update_interval": 0,
         "auto_restart": "yes",
         "remote_conf": "yes",
         "crypto_method": "aes",
         "server": [
            {
               "address": "10.2.0.10",
               "port": 1514,
               "max_retries": 5,
               "retry_interval": 10,
               "protocol": "tcp"
            }
         ],
         "enrollment": {
            "enabled": "yes",
            "delay_after_enrollment": 20,
            "port": 1515,
            "agent_name": "Rock",
            "group": "default",
            "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH",
            "auto_method": "no"
         }
      }
   },
   "error": 0
}
```

</details>

<details>
<summary>Tester @JuanCabrera - <code>DEB / Packages</code> :green_circle: </summary>

```
2022/05/31 16:37:24 wazuh-agentd: INFO: Trying to connect to server (10.0.0.1:1514/tcp).
2022/05/31 16:37:24 wazuh-agentd: INFO: (4102): Connected to the server (10.0.0.1:1514/tcp).
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client?pretty=true" -H  "Authorization: Bearer $token"
{
   "data": {
      "client": {
         "config-profile": "ubuntu, ubuntu16, ubuntu16.04",
         "notify_time": 10,
         "time-reconnect": 60,
         "force_reconnect_interval": 0,
         "ip_update_interval": 0,
         "auto_restart": "yes",
         "remote_conf": "yes",
         "crypto_method": "aes",
         "server": [
            {
               "address": "10.0.0.1",
               "port": 1514,
               "max_retries": 5,
               "retry_interval": 10,
               "protocol": "tcp"
            }
         ],
         "enrollment": {
            "enabled": "yes",
            "delay_after_enrollment": 20,
            "port": 1515,
            "agent_name": "Rock",
            "group": "default",
            "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH",
            "auto_method": "no"
         }
      }
   },
   "error": 0
}
```

</details>

## (TC_4): Set up connection with two managers, one with IP and one with FQDN, with enrollment in between

<details>
<summary>Tester @Vikman - <code>Ubuntu / Sources / AddressSanitizer</code> :green_circle: </summary>

```
{"data": {"client": {"notify_time": 10, "time-reconnect": 60, "force_reconnect_interval": 0, "ip_update_interval": 0, "auto_restart": "yes", "remote_conf": "yes", "crypto_method": "aes", "server": [{"address": "impish/192.168.33.12", "port": 1514, "max_retries": 5, "retry_interval": 10, "protocol": "tcp"}, {"address": "192.168.33.12", "port": 1514, "max_retries": 5, "retry_interval": 10, "protocol": "tcp"}], "enrollment": {"enabled": "yes", "delay_after_enrollment": 20, "port": 1515, "group": "default", "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH", "auto_method": "no"}}}, "error": 0}
```

</details>


<details>
<summary>Tester @Chema - <code>CentOS / Sources / Valgrind</code> :green_circle: </summary>

```
2022/05/31 18:12:37 wazuh-agentd[9892] start_agent.c:420 at agent_setup_hostnames(): INFO: Server hostname resolved: ubuntu/192.168.1.190
2022/05/31 18:12:37 wazuh-agentd[9892] start_agent.c:423 at agent_setup_hostnames(): INFO: Server IP Address: 1.1.1.1
2022/05/31 18:12:37 wazuh-agentd[9892] msgs.c:83 at OS_StartCounter(): DEBUG: OS_StartCounter: keysize: 1
2022/05/31 18:12:37 wazuh-agentd[9892] msgs.c:126 at OS_StartCounter(): DEBUG: Assigning counter for agent localhost.localdomain: '0:6642'.
2022/05/31 18:12:37 wazuh-agentd[9892] msgs.c:121 at OS_StartCounter(): DEBUG: Assigning sender counter: 0:1528
2022/05/31 18:12:37 wazuh-agentd[9892] msgs.c:140 at OS_StartCounter(): DEBUG: Stored counter.
2022/05/31 18:12:37 wazuh-agentd[9892] start_agent.c:242 at w_agentd_keys_init(): INFO: Using AES as encryption method.
2022/05/31 18:12:38 wazuh-agentd[9892] start_agent.c:98 at connect_server(): INFO: Trying to connect to server (ubuntu/192.168.1.190:1514/tcp).
2022/05/31 18:12:38 wazuh-agentd[9892] start_agent.c:361 at agent_handshake_to_server(): INFO: (4102): Connected to the server (ubuntu/192.168.1.190:1514/tcp).
```

Valgrind

```
==9892== LEAK SUMMARY:
==9892==    definitely lost: 0 bytes in 0 blocks
==9892==    indirectly lost: 0 bytes in 0 blocks
==9892==      possibly lost: 1,088 bytes in 4 blocks
==9892==    still reachable: 51,009 bytes in 48 blocks
==9892==         suppressed: 0 bytes in 0 blocks
==9892== Reachable blocks (those to which a pointer was found) are not shown.
==9892== To see them, rerun with: --leak-check=full --show-leak-kinds=all
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client" -H "Authorization: Bearer $TOKEN"
{"data": {"client": {"notify_time": 10, "time-reconnect": 60, "force_reconnect_interval": 0, "ip_update_interval": 0, "auto_restart": "yes", "remote_conf": "yes", "crypto_method": "aes", "server": [{"address": "ubuntu/192.168.1.190", "port": 1514, "max_retries": 5, "retry_interval": 10, "protocol": "tcp"}, {"address": "1.1.1.1", "port": 1514, "max_retries": 5, "retry_interval": 10, "protocol": "tcp"}], "enrollment": {"enabled": "yes", "delay_after_enrollment": 20, "port": 1515, "group": "default", "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH", "auto_method": "no"}}}, "error": 0}
```

</details>


<details>
<summary>Tester @Marcel - <code>CentOS / Packages</code> :green_circle: </summary>

```
2022/05/31 16:50:38 wazuh-agentd: INFO: Server hostname resolved: centosmanager/10.2.0.10
2022/05/31 16:50:38 wazuh-agentd: INFO: Server IP Address: 192.168.33.12
2022/05/31 16:50:38 wazuh-agentd: INFO: Using AES as encryption method.
2022/05/31 16:50:38 wazuh-agentd: INFO: Trying to connect to server (centosmanager/10.2.0.10:1514/tcp).
2022/05/31 16:50:38 wazuh-agentd: INFO: (4102): Connected to the server (centosmanager/10.2.0.10:1514/tcp).
```

Expected config applied

```
# curl -k -X GET "https://localhost:55000/agents/001/config/agent/client?pretty=true" -H  "Authorization: Bearer $TOKEN"
{
   "data": {
      "client": {
         "notify_time": 10,
         "time-reconnect": 60,
         "force_reconnect_interval": 0,
         "ip_update_interval": 0,
         "auto_restart": "yes",
         "remote_conf": "yes",
         "crypto_method": "aes",
         "server": [
            {
               "address": "centosmanager/10.2.0.10",
               "port": 1514,
               "max_retries": 5,
               "retry_interval": 10,
               "protocol": "tcp"
            },
            {
               "address": "192.168.33.12",
               "port": 1514,
               "max_retries": 5,
               "retry_interval": 10,
               "protocol": "tcp"
            }
         ],
         "enrollment": {
            "enabled": "yes",
            "delay_after_enrollment": 20,
            "port": 1515,
            "group": "default",
            "ssl_cipher": "HIGH:!ADH:!EXP:!MD5:!RC4:!3DES:!CAMELLIA:@STRENGTH",
            "auto_method": "no"
         }
      }
   },
   "error": 0
}
```

</details>

<details>
<summary>Tester @JuanCabrera - <code>DEB / Packages</code> :yellow_circle: </summary>

Pending to do...

</details>

**Comment by juliamagan** [comment]
## (System Tests): Enrollment

<details><summary>Launched locally by @juliamagan 🟢 </summary>

<h4>Enrollment test</h4>

- QA branch: **`4.3`**
- Wazuh branch: **`4.3.3`**
- Command to prepare the environment: **`ansible-playbook -i inventory.yml playbook.yml --extra-vars='{"wazuh_branch": "4.3.3"}'`**
- Test: **`test_cluster/test_agent_enrollment`**

|  Report  | Result     | Notes
|---    |---    |---    |
| [R1](https://github.com/wazuh/wazuh-qa/files/8812213/R1-test-system-enrollment.zip)  | 🟢  | - |
| [R2](https://github.com/wazuh/wazuh-qa/files/8812212/R2-test-system-enrollment.zip)| 🟢 | -| 
| [R3](https://github.com/wazuh/wazuh-qa/files/8812210/R3-test-system-enrollment.zip) | 🟢 | -| 

</details>
